csrc/preseg_passes/allocation_order_inference.cpp

```diff
@@ -77,6 +77,72 @@ void AllocationOrderInferencer::handle(UnaryOp* op) {
   propagateAllocationOrder(in, out);
 }
 
+// BroadcastOp propagation:
+//   1. preserves all allocation order of input iterdomain;
+//   2. stacks all added broadcast iter domain on outputs as outer dimensions in
+//   their natural position
+//
+// e.g.
+//   TV0 rfactor dom [i0, i1, i2] @ stride order {0, 2, 1}
+//    |    alloc dom [i0, i2, i1]
+//    |
+//    |
+//    BroadcastOp
+//    |
+//    v
+//   TV1 rfactor dom [i0, b3, i1, i2, b4]
+//         alloc dom [b3, b4, i0, i2, i1]  *see note 1
+//                     1,  4,
+//                             0,  3,  2   *see note 2
+//   note 1:
+//       keeping the alloc domain from input [i0, i2, i1]
+//       stack all broadcast in rfactor [b3, b4].
+//       concat([b3, b4], [i0, i2, i1])
+//
+//   note 2:
+//       computing the new output allocation order
+//       We'll scan through the rfactor domain of output where:
+//       a. insert any broadcast iterdomain index as we encounter them
+//       b. adjust the index of entry from input's rfactor domain
+//
+//   so output TV1 will have stride order {1, 4, 0, 3, 2}
+void AllocationOrderInferencer::handle(BroadcastOp* op) {
+  TensorView* out = dynamic_cast<TensorView*>(op->out());
```

Comment:
```suggestion
  auto* out = dynamic_cast<TensorView*>(op->out());
```

`dynamic_cast<TensorView*>` already spells out the type, so using `auto*` is fine and encouraged.